### ISSUE TITLE
Segfault when using the new deconjugation matcher

### ISSUE THREAD

**Issue by Calvin-Xu** [ripose-jp/Memento#229]
For a minimal reproducible example on my setup, simply scanning the に in

```一緒に```

triggers a segfault in Memento, as do some other inputs. I wonder if this can be reproduced elsewhere.

On another note, sorry I have not been using Memento lately and only getting around seeing the newer features, both due to other things that cause me to have dropped Anki for almost a year by now and due to mostly graduating from Memento and oftentimes just use a regular player and not on the desktop, which is a bittersweet feeling.

**Comment by ripose-jp** [comment]
Thanks for reporting this. I can't reproduce this on my setup. If I had to guess you have some dictionary installed that has some sort of data that leads to an out of bounds memory access somewhere in the deconjugator algorithm. Could you give me a list of dictionaries that you have installed and some guidance on where I can find them?

No need to apologize for not using Memento. You've done so much for this project and I'm very grateful. Hearing that you've been able to mostly move on is heartening because in that long term that's point. Good job.

**Comment by Calvin-Xu** [comment]
Thank you. I have quite a long list of dictionaries:

<img width="212" alt="image" src="https://github.com/ripose-jp/Memento/assets/28978265/27457b83-00cb-4df2-a4fa-eb8c997922ba">

if it helps, I can arrange to send my `dictionaries.sqlite` in a way that works for you

**Comment by aecsocket** [comment]
I can confirm this crash happens for me as well with a
```
(Paused) AV: 00:00:10 / 00:24:11 (1%) A-V:  0.000fish: Job 1, 'memento' terminated by signal SIGSEGV (Address boundary error)
```

Specifically when using these two enabled dictionaries:

![image](https://github.com/ripose-jp/Memento/assets/43144841/17cba659-3514-42c1-87ac-c1d6a80b1db8)

And in this text:

```
忘れたい思い出が染みついた場所だから
```

Hovering over `思い`

![image](https://github.com/ripose-jp/Memento/assets/43144841/88e6dcbf-9cb5-4ec9-8967-c440a9967113)

reliably causes a segfault. The screenshot shows the last frame rendered before it crashed, but I was hovering over the character right before `出`.

(Side note: these badges in the dictionary popup seem way too big, compared to Yomitan's popup at least)

**Comment by ripose-jp** [comment]
I was able to reproduce it using some of the aecsocket's dictionaries and his example text. There was a specific condition that could cause strict weak ordering in the comparator function. I think I've fixed it, and will try to get a new release out as soon as I verify that nothing bad happens to the ordering of terms.

If either of you can reproduce this with the new code. Let me know and feel free to reopen this if it's closed by then.

@aecsocket 
>(Side note: these badges in the dictionary popup seem way too big, compared to Yomitan's popup at least)

That's a bug in Jitendex as far as I can tell. Just update it and it should fix it. This was also covered in #223.

**Comment by ripose-jp** [comment]
I don't want to make a new release until I know this stable. Let me know if this solves your issues, it would be a big help.

**Comment by aecsocket** [comment]
Just built `bd830990784b81f80ff4e3d5367c2a9514b910fc` and tested out the same text with all dictionaries enabled. It didn't crash this time, so it looks like it's fixed.

Also tried with an updated Jitendex and it fixed the big badges issue.

**Comment by Calvin-Xu** [comment]
My preliminary usage did not cause any crashes, so it seems fixed.

**Comment by ripose-jp** [comment]
Thanks everyone's help. I've released v1.4.1 with the fix for this bug.
